scripts/xdcc_autoget.pl

```diff
@@ -0,0 +1,711 @@
+my $botcounter = 0;	
+my $packcounter = 0;
+my $episode = 1;
+
+my $server;		#current server
+
+sub ag_init		#init system
+{
+	Irssi::print "AG | Autoget initiated";
+	Irssi::print "AG | /ag_help for help";
+	&ag_initserver;
+	&ag_getbots;
+	my $m = "";
+	foreach my $n (@bots)
+	{
+		$m = $m . $n . ", ";
+	}
+	Irssi::print "AG | Bots: " . $m;
+	$m = "";
+	&ag_getterms;
+	foreach my $n (@terms)
+	{
+		$m = $m . $n . ", ";
+	}
+	Irssi::print "AG | Terms: " . $m;
+	if ($episodicflag)
+	{
+		Irssi::print "AG | Episodic: Yes";
+		Irssi::print "AG | Preffered format: $format";
+	}
+	else {Irssi::print "AG | Episodic: No";}
+	Irssi::print "AG | Data folder: $folder";
+}
+
+sub ag_initserver	#init server
+{
+	$server = " ";
+	$server = Irssi::active_server();	#keep trying to get server until it works, then continue after 5 seconds
+	if ($server !~ m/^Irssi::Irc::Server=HASH/) {Irssi::timeout_add_once(1000, sub {&ag_initserver;} , []);}
+	else {Irssi::timeout_add_once(5000, sub {&ag_run;} , []);}
+}
+
+sub ag_help
+{
+	Irssi::print "for this help    : ag_help";
+	Irssi::print "to run           : ag_run";
+	Irssi::print "to halt at next  : ag_stop";
+	Irssi::print "to set the server: ag_server";
+	Irssi::print "to add a bot     : ag_botadd BOT1 BOT2 *** BOTN";
+	Irssi::print "to remove a bot  : ag_botrem BOT1 BOT2 *** BOTN";
+	Irssi::print "to add string    : ag_add \"[TEXT STRING OF SEARCH]\",\"[ETC]\",***,\"[ETC]\"";
+	Irssi::print "to remove strings: ag_rem \"[TEXT STRING OF SEARCH]\",\"[ETC]\",***,\"[ETC]\"";
+	Irssi::print "to clear cache   : ag_clearcache";
+	Irssi::print "ag_next_delay            : delay between full transfers";
+	Irssi::print "ag_dcc_closed_retry_delay: delay after premature transfer";
+	Irssi::print "ag_bot_delay             : max time to wait for the bot to respond";
+	Irssi::print "ag_interrun_delay        : delay (in minutes, the rest seconds) between finishing a round and starting another";
+	Irssi::print "ag_autorun               : whether to run on startup";
+	Irssi::print "ag_episodic              : search ep 1, if found then search ep 2. Use for series if bot limits search results. Results may vary depending on bot and search term";
+	Irssi::print "ag_xdcc_send_prefix      : the xdcc message before the pack #";
+	Irssi::print "ag_xdcc_cancel_prefix    : the xdcc message to cancel a transfer";
+	Irssi::print "ag_xdcc_find_prefix      : the xdcc message before the search term";
+	Irssi::print "ag_format                : universal string appended to the end of each search in episodic. Use if more than one format exists";
+	Irssi::print "ag_folder                : Location for data files. ~/.irssi/ reccomended";
+}
+
+sub ag_server	#sets the current server
+{
+	$server = Irssi::active_server();	#keep trying to get server until it works
+	if ($server !~ m/^Irssi::Irc::Server=HASH/) {Irssi::timeout_add_once(1000, sub {&ag_server;} , []);}	
+}
+
+sub ag_getbots		#reads in bot list
+{
+	open(bots, "<", $botsfilename);
```

Comment:
this is not wrong, just a recommendation: Modern Perl suggests to use `open(my $bots, ...` instead of bare-word file-handles.